### PR TITLE
Optimize CDF Calculation and Convert NumPy Arrays to Tensors in Benchmark

### PR DESCRIPTION
**Pull request by yalsaffar**
### PR Description

This PR addresses the first step in making AEPsych's functions consistently return PyTorch tensors and expect tensors as input, improving compatibility with GPUs and reducing redundant conversions between NumPy arrays and PyTorch tensors(partially solving #365).

#### Key changes include:

1. **Conversion of `np.arrays` to tensors** in the following files:
   - **`aepsych/models/base.py`**: 
     - Refactored the `p_below_threshold` method to operate fully with PyTorch tensors.
     - Replaced `norm.cdf()` with `torch.distributions.Normal(0, 1).cdf()` for better GPU compatibility.

   - **`aepsych/benchmark/problem.py`**:
     - Significant changes made to ensure consistent use of tensors across the pipeline.
     - The result of `f_threshold()` now directly returns a PyTorch tensor, ensuring consistency.
     - Additionally, used `detach().cpu().numpy()` in places where the `super().evaluate()` method returns float values, ensuring compatibility.

2. **Updates in `aepsych/tests/test_benchmark.py`**:
   - Migrated all operations from NumPy to PyTorch.
   - This includes calculations for Brier score and misclassification error, now utilizing `torch.mean()`, `torch.square()`, `torch.isclose()`, and `torch.all()` to fully align with tensor operations.

#### Stability:
All test cases have passed successfully in the workflow.

### CODE REVIEW

File: aepsych/models/base.py
Comment (JasonKChow) [inline aepsych/models/base.py:1]: Assuming more change will come to this file but leaving it for another PR.

**Review by JasonKChow — CHANGES_REQUESTED** [review]
There's still some functions that return numpy arrays (despite the documentation specifically saying otherwise). Example: in f_true method of Problem class.

Some typehints are wrong, example f_true method again.

The evaluate method can be rewritten entirely to use torch. Replace pearsonr with corrcoef for torch, you'll need to reformat the data.

File: aepsych/benchmark/problem.py [hunk 1]
```diff
@@ -233,12 +234,13 @@ def f_threshold(self, model=None) -> torch.Tensor:
             inverse_torch = model.likelihood.objective.inverse
 
             def inverse_link(x):
-                return inverse_torch(torch.tensor(x))
+                return inverse_torch(torch.tensor(x).clone().detach())
```
Comment (JasonKChow) [inline aepsych/benchmark/problem.py:237]: Why is this cloned and detached? Also why is it converted to tensor if  we already assume that self.thresholds will be a tensor?

Comment (yalsaffar) [reply aepsych/benchmark/problem.py:237]: You're right, there's no need to convert it again, and cloning and detaching are also unnecessary. I'll remove those, test it, and commit the changes.

**Review by JasonKChow — APPROVED** [review]
